### DISCUSSION TITLE
A zlib compress / decompress utility - store multiple files / directory structure in a single compressed blob.

### DISCUSSION THREAD

**Discussion by sethstenzel** [micropython/micropython#11117] [Show and tell]
This is a set of utilities for compressing files or a folder with zlib, then decompressing on the device. A simple alternative to zip files for me.
This will allow multiple files to be combined and then compressed with zlib. A single file can be created, and then an index file can be used later to name the end to unpack the files. The directory structure is maintained in the index file. The index file stores a relative file path, the length of the file, and then an offset of where the file starts.
Here is an example of the index file.
```
{
    ".\\zlib_compressor.py": [0, 3288],
    ".\\zlib_extractor.py": [3288, 3324]
}
```

This is very basic in implementation, and I have not refactored or tried cleaning things up.

Sharing it here in case anyone has a use for it. I've used it in a few instances to reduce file size, download and update the package, and deploy it on the device.

[`https://github.com/seth-c-stenzel/zlib_utils`](https://github.com/seth-c-stenzel/zlib_utils)

**Comment by jonnor** [comment]
Nice, thank you for sharing!

**Comment by jonnor** [comment]
There is now also a port of `zipfile` from CPython to MicroPython at https://github.com/jonnor/micropython-zipfile - for those who need standard ZIP archives
